File: backend-springboot/todo/src/main/java/net/indralight/todo/controller/UserController.java
package net.indralight.todo.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
//import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
//import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import io.swagger.annotations.ApiOperation;
import io.swagger.annotations.ApiResponse;
import io.swagger.annotations.ApiResponses;
import net.indralight.todo.dto.KakaoOAuthDTO;
import net.indralight.todo.dto.ResponseDTO;
import net.indralight.todo.dto.UserRequestDTO;
import net.indralight.todo.dto.UserResponseDTO;
import net.indralight.todo.model.UserEntity;
import net.indralight.todo.security.TokenProvider;
import net.indralight.todo.service.UserService;

//import io.swagger.annotations.Api;
//import io.swagger.annotations.ApiImplicitParam;
//import io.swagger.annotations.ApiOperation;
//import io.swagger.annotations.ApiResponse;
//import io.swagger.annotations.ApiResponses;

//@Api(tags = { "인증 처리 Controller" })
@RestController
@RequestMapping("/auth")
public class UserController {

    @Autowired
    private UserService userService;

    @Autowired
    private TokenProvider tokenProvider;

    // private PasswordEncoder passwordEncoder = new BCryptPasswordEncoder();

    @ApiOperation(value = "유저 등록")
    // @ApiImplicitParam(name = "userDTO", dataType = "UserDTO")
    @ApiResponses({ @ApiResponse(code = 200, response = UserEntity.class, message = "성공"),
            @ApiResponse(code = 400, response = ResponseDTO.class, message = "실패") })
    @PostMapping("/signup")
    public ResponseEntity<?> signup(@RequestBody UserRequestDTO userRequestDTO) {
        try {
            UserEntity user = UserEntity.builder()
                    .email(userRequestDTO.getEmail())
                    .nick(userRequestDTO.getNick())
                    .password(userRequestDTO.getPassword())
                    .provider("local")
                    .build();

            UserEntity registeredUser = userService.create(user);

            return ResponseEntity.ok()
                    .body(registeredUser);
        } catch (Exception e) {
            ResponseDTO responseDTO = ResponseDTO.builder()
                    .error(e.getMessage())
                    .build();
            return ResponseEntity.badRequest()
                    .body(responseDTO);
        }
    }

    @ApiOperation(value = "유저 로그인")
    // @ApiImplicitParam(name = "userDTO", dataType = "UserDTO")
    @ApiResponses({ @ApiResponse(code = 200, response = UserResponseDTO.class, message = "성공"),
            @ApiResponse(code = 400, response = ResponseDTO.class, message = "실패") })
    @PostMapping("/signin")
    public ResponseEntity<?> signin(@RequestBody UserRequestDTO userRequestDTO) {
//        UserEntity user = userService.getByCredentials(userDTO.getEmail(), userDTO.getPassword(),
//                passwordEncoder);
        UserEntity user = userService.getByCredentials(userRequestDTO.getEmail(),
                userRequestDTO.getPassword());

        if (user != null) {
            final String token = tokenProvider.create(user);
            final UserResponseDTO userResponseDTO = UserResponseDTO.builder()
                    .token(token)
                    .user(user)
                    .build();

            return ResponseEntity.ok()
                    .body(userResponseDTO);
        } else {
            ResponseDTO responseDTO = ResponseDTO.builder()
                    .error("Login failed.")
                    .build();

            return ResponseEntity.badRequest()
                    .body(responseDTO);
        }

    }

    @ApiOperation(value = "카카오 로그인")
    // @ApiImplicitParam(name = "userDTO", dataType = "UserDTO")
    @ApiResponses({ @ApiResponse(code = 200, response = UserResponseDTO.class, message = "성공"),
            @ApiResponse(code = 400, response = ResponseDTO.class, message = "실패") })
    @PostMapping("/kakao")
    public ResponseEntity<?> kakao(@RequestBody KakaoOAuthDTO kakaoOAuthDTO) {
        UserEntity user = userService.getByKakao(kakaoOAuthDTO);

        if (user != null) {
            final String token = tokenProvider.create(user);
            final UserResponseDTO userResponseDTO = UserResponseDTO.builder()
                    .token(token)
                    .user(user)
                    .build();

            return ResponseEntity.ok()
                    .body(userResponseDTO);
        } else {
            ResponseDTO responseDTO = ResponseDTO.builder()
                    .error("Login failed.")
                    .build();

            return ResponseEntity.badRequest()
                    .body(responseDTO);
        }
    }
}

File: backend-aspnet/Todo/Controllers/UserController.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;
using Todo.Entity;
using Todo.Model;
using Todo.Service;

namespace Todo.Controllers
{
    [ApiController]
    [Route("auth")]
    public class UserController : ControllerBase
    {
        private readonly ILogger<UserController> _logger;
        private readonly UserService _userService;
        private readonly JwtManager _jwtManager;

        public UserController(ILogger<UserController> logger, UserService userService, JwtManager jwtManager)
        {
            _logger = logger;
            _userService = userService;
            _jwtManager = jwtManager;
        }

        [HttpPost("signup")]
        public async Task<ActionResult<User>> SignUp(UserRequest request)
        {
            const string provider = "local";
            try
            {
                var user = new User
                {
                    Email = request.Email,
                    Nick = request.Nick ?? "",
                    Password = request.Password,
                    Provider = provider,
                };
                await _userService.Create(user);
                return Ok(user);
            }
            catch (Exception e)
            {
                return BadRequest(new { error = e.Message });
            }
        }

        [HttpPost("signin")]
        public async Task<ActionResult<UserResponse>> SignIn(UserRequest request)
        {
            var user = await _userService.SignInLocal(request.Email, request.Password);
            if (user != null)
            {
                var token = _jwtManager.Generate(user);
                var response = new UserResponse
                {
                    Token = token,
                    User = user,
                };
                return Ok(response);
            }
            else
            {
                return BadRequest(new { error = "Login failed." });
            }
        }

        [HttpPost("kakao")]
        public async Task<ActionResult<UserResponse>> Kakao(KakaoOAuthRequest request)
        {
            var user = await _userService.GetByKakao(request);

            if (user != null)
            {
                var token = _jwtManager.Generate(user);
                var response = new UserResponse
                {
                    Token = token,
                    User = user,
                };
                return Ok(response);
            }
            else
            {
                return BadRequest(new { error = "Login failed." });
            }
        }
    }
}